lib/elixir/lib/calendar/duration.ex

```diff
@@ -0,0 +1,194 @@
+defmodule Duration do
+  @moduledoc """
+  Struct and functions for handling durations.
+
+  A `Duration` struct represents a collection of time scale units,
+  allowing for manipulation and calculation of durations.
+
+  Date and time scale units are represented as integers, allowing for both positive and negative values.
+
+  Microseconds are represented using a tuple `{microsecond, precision}`.
+  This ensures compatibility with other calendar types implementing time, such as `Time`, `DateTime`, and `NaiveDateTime`.
+  """
+
+  @moduledoc since: "1.17.0"
+
+  defstruct year: 0,
```

Comment:
```suggestion
  @derive {Inspect, optional: [:year, :month, :week, :day, :hour, :minute, :second, :microsecond]}
  defstruct year: 0,
```

This way, we only show the fields that do not match the default value, making the default inspect much more readable. WDYT?

Reply:
I think that's helpful! Done in https://github.com/elixir-lang/elixir/pull/13385/commits/f69baf1277e4587d491224dde8363d8c950f766a